### ISSUE TITLE
Using the `--filter` option in the root directory will pass an empty string to `micromatch`

### ISSUE THREAD

**Issue by ever0de** [pnpm/pnpm#5051]
### Code to reproduce the issue: 

- <https://github.com/ever0de/pnpm-docker-root-bug>

### Expected behavior: 

- Commands using the `--filter` option used in docker should work normally.
- Or a warning message is displayed telling you not to use it in root

### Actual behavior:

- `pnpm: Expected pattern to be a non-empty string` error message like this is thrown.

### Additional information:

 - `node -v` prints: `v16.15.0`
 - Windows, macOS, or Linux?: `macOS` (laptop), `Linux` (docker)

### Other..

It seems more like an improvement than a bug.
I wish the error message was a bit more friendly.

**Comment by jclaessens97** [comment]
I had the same issue (`pnpm v7.9.0`). Only reason I found the fix was because of this issue 😅

**Comment by ever0de** [comment]
> I had the same issue (`pnpm v7.9.0`). Only reason I found the fix was because of this issue 😅

Did you solve the problem? 
Note) https://github.com/ever0de/pnpm-docker-root-bug#how-to-fix

**Comment by jclaessens97** [comment]
Yes, I saw the fix in the repo, thanks. But I agree the error message could me more clear.

**Comment by ever0de** [comment]
> Yes, I saw the fix in the repo, thanks. But I agree the error message could me more clear.

Glad to be of help 👍

**Comment by Kenjirou** [comment]
I actually didn't understand what is the fix. I have the same problem with the latest (`7.28.0`) version.
We have `CMD ["pnpm", "-F", "apps/web", "start"]` in the Dockerfile and it leads to "ERROR  Expected pattern to be a non-empty string".

**Comment by ever0de** [comment]
> I actually didn't understand what is the fix. I have the same problem with the latest (`7.28.0`) version. We have `CMD ["pnpm", "-F", "apps/web", "start"]` in the Dockerfile and it leads to "ERROR  Expected pattern to be a non-empty string".

In your docker file, create a WORKDIR, but not from your current location (i.e. root).
https://github.com/ever0de/pnpm-docker-root-bug#how-to-fix

**Comment by Kenjirou** [comment]
Now I got it, thanks! But it's not an actual fix it's just a workaround to run scripts from the single app directory as I see.

**Comment by ever0de** [comment]
Yes, even if it's the root, you have to handle it without passing the empty string.

**Comment by zkochan** [comment]
This targets a part of pnpm that no longer exists, so closing.

pnpm 12's CLI is native code and no longer routes filters through micromatch. Filters at the filesystem root behave the same as anywhere else.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
